source/module_psi/psi.h

```diff
@@ -188,6 +196,9 @@ class Psi
         assert(ik>=0 && ik<this->nk);
 		assert(ibands>=0 && ibands<this->nbands);	
         assert(ibasis>=0 && ibasis<this->nbasis);
+        #if ((defined __CUDA) || (defined __ROCM))
+            ModuleBase::WARNING_QUIT("Psi operator ()", "GPU psi cannot fetch value by an overloaded operator ()!");
+        #endif 
```

Comment:
Maybe using a macro for this repeated code would be better.

Reply:
Maybe the next PR.